### PR TITLE
Handle eth_signTypedData

### PR DESCRIPTION
**Pull request by ukstv**
Allow HookedWalletSubprovider to `signTypedData` as per https://github.com/ethereum/EIPs/pull/712

### CODE REVIEW

**Comment by flyswatter** [comment]
Also, the discussion in [EIP 712](https://github.com/ethereum/EIPs/pull/712#issuecomment-331875764) has leaned towards naming this method `eth_signTypedData` instead, so could you make that change too?

**Comment by ukstv** [comment]
@flyswatter What about now?

File: test/wallet.js [hunk 1]
```diff
@@ -250,6 +250,21 @@ recoverTest({
   addressHex: '0xbe93f9bacbcffc8ee6663f2647917ed7a20a57bb',
 })
 
+signatureTest({
+  testLabel: 'kumavis fml manual test I',
```
Comment (kumavis) [inline test/wallet.js]: new test label please

File: test/wallet.js [hunk 2]
```diff
@@ -250,6 +250,21 @@ recoverTest({
...
+  message: [
+    {
+      type: 'string',
+      name: 'message',
+      value: 'Hi, Alice!'
+    }
+  ],
+  signature: '0xb2c9c7bdaee2cc73f318647c3f6e24792fca86a9f2736d9e7537e64c503545392313ebbbcb623c828fd8f99fd1fb48f8f4da8cb1d1a924e28b21de018c826e181c',
+  addressHex: '0xbe93f9bacbcffc8ee6663f2647917ed7a20a57bb',
+  privateKey: new Buffer('6969696969696969696969696969696969696969696969696969696969696969', 'hex'),
```
Comment (kumavis) [inline test/wallet.js]: too erotic 👎 
er wait 🤔 
sufficiently erotic 👍

File: test/wallet.js [hunk 3]
```diff
@@ -313,7 +328,7 @@ function signatureTest({ testLabel, method, privateKey, addressHex, message, sig
 
   var payload = {
     method: method,
-    params: [addressHex, message],
+    params: [message, addressHex],
```
Comment (kumavis) [inline test/wallet.js]: this seems out of bounds -- is this the correct param ordering for all signing methods?

Comment (danfinlay) [reply test/wallet.js]: Yes, [message, address] is the standard for the signing methods. We support the other order when concretely detected to support a time when it was less standard.

Comment (ukstv) [reply test/wallet.js]: Timing for this line and warning in [HookedWallet](https://github.com/MetaMask/provider-engine/blob/master/subproviders/hooked-wallet.js#L139) suggested this is the right way.

**Comment by ukstv** [comment]
@kumavis So true. Changed the label.

File: test/wallet.js (hunk 3 above)
Comment (kumavis) [reply test/wallet.js]: ok but what about `eth_sign`?
I would expect this to break `eth_sign` tests (tho it doesnt seem to)

Comment (kumavis) [reply test/wallet.js]: I see, we dont use this helper for `eth_sign` tests :+1:

**Comment by petejkim** [comment]
This implementation is not up to date and generates different signatures, is anyone already working on updating it?

**Comment by mtabasco** [comment]
Hi! Why this implementation doesnt support JSON schema specified in https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md#specification-of-the-eth_signtypeddata-json-rpc? Is there a plan to implement same behaviour of `eth_signTypedData_v3`? Good work, thanks!

**Pull request merged**